core/src/main/scala/scalaz/ImmutableArray.scala

```diff
@@ -31,7 +30,7 @@ sealed abstract class ImmutableArray[+A] {
   def ++[B >: A: ClassTag](other: ImmutableArray[B]): ImmutableArray[B]
 }
 
-sealed abstract class ImmutableArrayInstances {
```

Comment:
I was wondering what the instances for `ImmutableArray` are for until I realized that scalaz has its own `ImmutableArray` type. Ours might still be renamed to it would resolve the ambiguity but may create another one (because that's why we want to remove it in the first place). See https://github.com/scala/bug/issues/10836.